Karma/Controllers/PostsController.cs

```diff
@@ -28,10 +28,22 @@ public PostsController(KarmaContext context, IWebHostEnvironment webHostEnvironm
         }
 
         // GET: Posts
-        public async Task<IActionResult> Index()
+        public async Task<IActionResult> Index(bool? isDonation)
         {
-            Console.WriteLine("test");
-            return View(await _context.Post.ToListAsync());
+            List<Post> posts;
+
+            if (isDonation == null)
+            {
+                ViewBag.Header = "All posts";
+                posts = await _context.Post.ToListAsync();
+            }
+            else
+            {
+                posts = await _context.Post.Where(p => p.IsDonation == isDonation).ToListAsync();
```

Comment:
What's the purpose of having parameter `bool? isDonation`?

Reply:
Do you mean why can it be nullable? It's because this same controller method applies to pages /Posts, /Posts?isDonation=true and /Posts?isDonation=false. So isDonation is null when calling this method for /Posts page.

The default pattern in `Startup.cs` might already catch that, but I'm not sure. I should probably test that.
```
endpoints.MapControllerRoute(name: "default",
                pattern: "{controller=Home}/{action=Index}/{id?}");
```